docs/firmware/golioth-firmware-sdk/light-db-state/guide-lightdb-observe.md

```diff
@@ -0,0 +1,100 @@
+---
+title: LightDB Observe
+sidebar_position: 3
+---
+
+Use a LightDB observe function to receive an update whenever a specific path
+value changes on the cloud. The observe request works in a similar way to the
+get request and much of the setup is the same. Data is received asynchronously
+and sent to a handler function registered when the observe request is first
+made.
+
+## Includes
+
+```c
+#include "golioth.h"
+```
+
+Including the `golioth.h` header file makes the Golioth API functions available
+to your program.
+
+## Asynchronous Observe
+
+The asynchronous function `golioth_lightdb_observe_async()` is a non-blocking
```

Comment:
This comment isn't relevant to this docs PR, but I was thinking about it just now: Should this function just be `golioth_lightdb_observe()`? All observations are asynchronous, and I don't know whether having the `_async` here is helpfully explicit, or if it implies that there is a corresponding `_sync` function somewhere.

Reply:
Every function name in the golioth_lightdb module ends with either `_sync` or `_async` so I think it makes sense to keep with that pattern.

I also like the explicit nature of the function name as it helps me remember what params will be needed (callbacks for async, timeouts for sync).